qiskit/compiler/assemble.py

```diff
@@ -276,13 +283,29 @@ def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
 
     qubit_lo_range = qubit_lo_range or getattr(backend_config, 'qubit_lo_range', None)
     meas_lo_range = meas_lo_range or getattr(backend_config, 'meas_lo_range', None)
-    rep_time = rep_time or getattr(backend_config, 'rep_times', None)
 
-    if isinstance(rep_time, list):
-        rep_time = rep_time[0]
+    dynamic_reprate_enabled = getattr(backend_config, 'dynamic_reprate_enabled', False)
 
+    rep_time = rep_time or getattr(backend_config, 'rep_times', None)
     if rep_time:
-        rep_time = int(rep_time * 1e6)
+        # deprecation warning for ``rep_time`` when dynamic rep rate is enabled
```

Comment:
```suggestion
```
this comment restates the code

Reply:
Addressed in https://github.com/Qiskit/qiskit-terra/pull/4539/commits/d192e28937f3e8826260746c7d083e272ffa2075.